src/graphics.cpp

```diff
@@ -250,6 +258,10 @@ void Graphics::Transition(TransitionType type, int duration, bool erase) {
 			else
 				screen1 = screen2;
 		}
+		
+		if (type == TransitionRandomBlocks) {
+			std::shuffle(randomBlocks.begin(), randomBlocks.end(), std::default_random_engine(next_fps_time));
```

Comment:
That's a clever solution!
For portability reasons (you can pass a seed via --seed command line argument) all random functions must use our shared RNG.
A suitable function here is probably Utils::GetRandomU32(), unfortunately the function is cpp-file static. You could add it to the utils.h header and then call it.